.evergreen/install-dependencies.sh

```diff
@@ -24,6 +24,9 @@ source ~/.cargo/env
 # Install nightly rustfmt
 rustup toolchain install nightly -c rustfmt
 
+# Pin clippy version to the latest Rust version. This should be updated with each new Rust release.
+rustup toolchain install 1.59.0 -c clippy
```

Comment:
I think we may need to update `check-clippy.sh` to select `1.59.0` as the toolchain to use. I think otherwise it'll just use stable by default.

Reply:
Removed this in favor of running `rustup default 1.59.0` in `check-clippy.sh`. I tested this out locally by changing the default to a few different versions and running `cargo clippy` and it seems to do the trick.